packages/grpc-js-core/src/client.ts

```diff
@@ -147,13 +146,14 @@ export class Client {
              metadata, options, callback));
     const call: CallStream =
         this.channel.createStream(method, metadata, options);
-    const emitter: ClientUnaryCall = new ClientUnaryCallImpl(call);
     const message: Buffer = serialize(argument);
     const writeObj: WriteObject = {message: message};
     writeObj.flags = options.flags;
     call.write(writeObj);
     call.end();
     this.handleUnaryResponse<ResponseType>(call, deserialize, callback);
+
+    const emitter: ClientUnaryCall = new ClientUnaryCallImpl(call);
```

Comment:
Why move this (and the others in the file) to later in the function?

Reply:
The `ClientUnaryCall` is made only to be passed to the client, and doesn't have to be a local variable at all (we can just do `return new ClientUnaryCallImpl(call);` instead). I guess this was a conservative step towards that approach. Do you think it's more readable with all of the object allocations together, or avoiding creating the `emitter` local variable entirely?

Reply:
For some reason I was thinking that you would want the `emitter` to exist before taking actions on the call, but everything is asynchronous anyway so that doesn't really matter. I think it makes sense to just remove the `emitter` variable.